Fix build with >=poppler-0.62.0, cleanup ifdef and raise minimum poppler version.

--- a/configure.ac	2017-12-12 18:45:07.000000000 +0100
+++ b/configure.ac	2018-01-04 23:57:16.410999542 +0100
@@ -10403,7 +10403,7 @@
     if test "$with_system_poppler" = "yes"; then
         AC_MSG_RESULT([external])
         SYSTEM_POPPLER=TRUE
-        PKG_CHECK_MODULES( POPPLER, poppler >= 0.12.0 )
+        PKG_CHECK_MODULES( POPPLER, poppler >= 0.21.1 )
         AC_LANG_PUSH([C++])
         save_CXXFLAGS=$CXXFLAGS
         save_CPPFLAGS=$CPPFLAGS
--- a/sdext/source/pdfimport/xpdfwrapper/pdfioutdev_gpl.cxx	2017-12-12 18:45:07.000000000 +0100
+++ b/sdext/source/pdfimport/xpdfwrapper/pdfioutdev_gpl.cxx	2018-01-05 00:29:16.541246738 +0100
@@ -32,13 +32,11 @@
 #pragma warning(push, 1)
 #endif
 
-// sigh, UTF8.h was removed in poppler-0.21.0 and put back in 0.21.1
 // FIXME: we can't use #if POPPLER_CHECK_VERSION(0, 21, 0) && !POPPLER_CHECK_VERSION(0, 21, 1)
 //        because the internal poppler does not provide poppler-version.h and the macro always returns 0
-#if POPPLER_CHECK_VERSION(0, 21, 1)
-#include "UTF8.h"
-#elif POPPLER_CHECK_VERSION(0, 21, 0)
-#include "UTF.h"
+// UTF8.h was renamed to UnicodeMapFuncs.h in poppler-0.62.0
+#if POPPLER_CHECK_VERSION(0, 62, 0)
+#include "UnicodeMapFuncs.h"
 #else
 #include "UTF8.h"
 #endif
@@ -918,7 +916,11 @@
             );
 
     // silence spurious warning
+#if POPPLER_CHECK_VERSION(0, 62, 0)
+    (void)&mapUTF16;
+#else
     (void)&mapUCS2;
+#endif
 
     char buf[9];
     for( int i=0; i<uLen; ++i )
